recipes/default.rb

```diff
@@ -44,38 +37,25 @@
   action :install
 end
 
-directory node['pgd']['pgd_path'] do
+python_webapp 'pgd_core' do
+  create_user true
   owner node['pgd']['user']
   group node['pgd']['group']
-  mode '0755'
-  action :create
-end
 
-git node['pgd']['pgd_path'] do
   repository node['pgd']['git']['repository']
   revision node['pgd']['git']['revision']
-  user node['pgd']['user']
-  group node['pgd']['group']
-end
-
-python_pip "#{node['pgd']['pgd_path']}/requirements.txt" do
-  options '-r'
-  virtualenv node['pgd']['virtualenv_path']
-  action :install
-end
 
-#::File.join(node['ganeti_webmgr']['config_dir'], 'config.yml')
-config_file = "#{node['pgd']['pgd_path']}/pgd/settings.py"
+  path node['pgd']['pgd_path']
+  virtualenv_path node['pgd']['virtualenv_path']
+  requirements_file 'requirements.txt'
 
-template config_file do
-  source 'django_settings.py.erb'
-  owner node['pgd']['user']
-  group node['pgd']['group']
-  mode '0644'
-  variables(
+  config_template 'django_settings.py.erb'
+  config_destination "#{ node['pgd']['pgd_path'] }/settings.py"
```

Comment:
Because secretly you should fear for your life everytime you call use `node`. It's basically a giant global variable that is never guaranteed to be in the same state when you call it as it will be at the end, and can run into really weird things like:

```ruby
resource "name" do
  attr node['foo']['bar'] # node['foo']['bar'] := 'foobar' here
end

node.override['foo']['bar'] = 'foobarbaz'
```

and then every subsequent call to `node['foo']['bar']` will return `"foobarbaz"` after that, but the resource attr will still receive `foobar`.